LiteCore/Database/TreeDocument.cc

```diff
@@ -589,7 +591,9 @@ namespace litecore {
             if (!body)
                 return false;
 
-            revidBuffer encodedNewRevID = generateDocRevID(body, _selected.revID, deletion);
+            DatabaseImpl* db = _collection->dbImpl();
+            revidBuffer encodedNewRevID = generateDocRevID(body, _selected.revID, deletion,
```

Comment:
Instead of adding a SharedKeys parameter to `generateDocRevID`, just make that method non-static, so it can get it itself. (It still makes sense for it to take the revID as a parameter, IMO.)

Reply:
@snej I just found out that we have duplicated code of generating rev id in 
   alloc_slice VectorRecord::generateRevID(Dict body, revid parentRevID, DocumentFlags flags)
Do I need to take care of it here?

Reply:
No. Version-vector revIDs are not affected by the content of the document, so they don't have this issue that we're fixing.